### ISSUE TITLE
Bug: Windows x64 quick start web template error

### ISSUE THREAD

**Issue by CoreyCole** [aws/aws-sam-cli#3946]
### Description:

Follwing the documentation [here](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/serverless-sam-cli-install-windows.html) I ran `sam init` and it threw the error:
```
Error: Can't find application template quick-start-web - check valid values in interactive init.
``` 

### Steps to reproduce:

```
sam init

        SAM CLI now collects telemetry to better understand customer needs.

        You can OPT OUT and disable telemetry collection by setting the
        environment variable SAM_CLI_TELEMETRY=0 in your shell.
        Thanks for your help!

        Learn More: https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/serverless-sam-telemetry.html


You can preselect a particular runtime or package type when using the `sam init` experience.
Call `sam init --help` to learn more.

Which template source would you like to use?
        1 - AWS Quick Start Templates
        2 - Custom Template Location
Choice: 1

Choose an AWS Quick Start application template
        1 - Hello World Example
        2 - Multi-step workflow
        3 - Serverless API
        4 - Scheduled task
        5 - Standalone function
        6 - Data processing
        7 - Infrastructure event management
        8 - Machine Learning
Template: 3

Which runtime would you like to use?
        1 - dotnet6
        2 - dotnetcore3.1
        3 - nodejs16.x
        4 - nodejs14.x
        5 - nodejs12.x
        6 - python3.9
        7 - python3.8
Runtime: 4

Based on your selections, the only Package type available is Zip.
We will proceed to selecting the Package type as Zip.

Based on your selections, the only dependency manager available is npm.
We will proceed copying the template using npm.

Would you like to enable X-Ray tracing on the function(s) in your application?  [y/N]: y
X-Ray will incur an additional cost. View https://aws.amazon.com/xray/pricing/ for more details

Project name [sam-app]:

Cloning from https://github.com/aws/aws-sam-cli-app-templates (process may take a moment)
2022-05-31 17:59:58,957 |
Cloning from https://github.com/aws/aws-sam-cli-app-templates (process may take a moment)
2022-05-31 18:00:02,025 | Clone error, attempting to use an old clone from a previous run
2022-05-31 18:00:02,025 | Template is not provided in context, skip adding project type metric
2022-05-31 18:00:02,026 | Sending Telemetry: {'metrics': [{'commandRun': {'requestId': '2a94d214-eb7b-410d-914e-f5582e523006', 'installationId': '53585486-101e-4a05-a49c-15c276a9dd16', 'sessionId': '98473969-3ad9-4888-8338-80c4d8a8c2d4', 'executionEnvironment': 'CLI', 'ci': False, 'pyversion': '3.8.8', 'samcliVersion': '1.51.0', 'awsProfileProvided': False, 'debugFlagProvided': True, 'region': '', 'commandName': 'sam init', 'duration': 19671, 'exitReason': 'InvalidInitTemplateError', 'exitCode': 1}}]}
2022-05-31 18:00:02,179 | Telemetry response: 200
Error: Can't find application template quick-start-web - check valid values in interactive init.
```

### Additional environment details (Ex: Windows, Mac, Amazon Linux etc)

1. OS: Windows x64
2. If using SAM CLI, `sam --version`: 1.51.0
3. AWS region:

**Comment by hawflau** [comment]
Hey @CoreyCole, the error was due to the limitation of 260 chars in a path in Windows. We will push a fix to enable this behaviour for the app templates repo.

Meanwhile, to workaround it, you can run this command before running `sam init`:
```
git config --global core.longpaths true
```
Note that this affects your global git settings, and it might have other drawback. So you might want to unset it after running `sam init`:
```
git config --global --unset core.longpaths
```

**Comment by ActionPace** [comment]
I had a previous issue today (but it was simply deleted with no comment) related to this where all the python templates were missing in the C:\Users\USER\AppData\Roaming\AWS SAM\aws-sam-cli-app-templates folder

Although the error I was getting was a git clone error during the sam init process and then it ended with "Error: Please verify your location.."

Even another git clone would not pull down the python templates so I ended up downloading the zipped repository and placing it as the new folder name above. As per the comment above, is there something not right about the git repository?

Aldo, do you think these two issues are related?

**Comment by ActionPace** [comment]
In regards to my issue, I tried the workaround above and seemed to have fixed the weird git status by typing:

git config --global core.longpaths true

When I reran the sam init --debug and chose option 3 for Serverless API (both python and nodejs) I got a huge dump of errors and it said something about a permission problem in C:\Users\David\AppData\Roaming\AWS SAM
I tried rerunning cmd as admin and rerunning the command but got the same errors..

I looked in C:\Users\David\AppData\Roaming\AWS SAM and saw the aws-sam-cli-app-templates folder appear and looked in there and it seems git was able to get all the files this time

I unset the git setting with: 
git config --global --unset core.longpaths

And now it seems I can run sam init normally now for option 3 Serverless API (both node and python)
I haven't tried the other options yet but this looks promising

Hopefully something can get fixed in the next release of sam cli

**Comment by qingchm** [comment]
I believe that this is fixed in the new 1.53.0 release. Please have a try and if you still have any questions or issues feel free to re-open the issue!
